packages/openjdk-21/build.sh

```diff
@@ -0,0 +1,102 @@
+TERMUX_PKG_HOMEPAGE=https://openjdk.java.net
+TERMUX_PKG_DESCRIPTION="Java development kit and runtime"
+TERMUX_PKG_LICENSE="GPL-2.0"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION=21.0.3+6
+TERMUX_PKG_REVISION=0
+TERMUX_PKG_SRCURL=https://github.com/openjdk/jdk21u/archive/refs/tags/jdk-${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_SHA256=75a6ca52299ee2f5b028a5a4acedd6139ae740cbf1af034e2d50489e74598dd3
+TERMUX_PKG_AUTO_UPDATE=false
+TERMUX_PKG_DEPENDS="libandroid-shmem, libandroid-spawn, libiconv, libjpeg-turbo, zlib"
+TERMUX_PKG_BUILD_DEPENDS="cups, fontconfig, libxrandr, libxt, xorgproto"
+# openjdk-21-x is recommended because X11 separation is still very experimental.
+TERMUX_PKG_RECOMMENDS="ca-certificates-java, openjdk-21-x, resolv-conf"
+TERMUX_PKG_SUGGESTS="cups"
+TERMUX_PKG_BUILD_IN_SRC=true
+TERMUX_PKG_HAS_DEBUG=false
+
+termux_step_pre_configure() {
+	unset JAVA_HOME
+
+	env -i PATH="$PATH" sudo apt update
+	env -i PATH="$PATH" sudo apt install openjdk-21-jdk-headless -y
```

Comment:
Adding openjdk-21-jdk-headless should be done in [setup-ubuntu.sh](https://github.com/termux/termux-packages/blob/master/scripts/setup-ubuntu.sh), in a separate commit so that we can cherry-pick it to master. 

openjdk-17 and openjdk-21 needs different java versions to build. Probably both packages should select the one they need in termux_step_pre_configure (by running a list of update-alternatives commands, unless ubuntu has some "java-selector" package I do not know about)

Reply:
looks like openjdk configure has adapted for debian distro:
```bash
configure: Found potential Boot JDK using JAVA_HOME
configure: Potential Boot JDK found at /usr/lib/jvm/java-17-openjdk-amd64 is incorrect JDK version (openjdk version "17.0.11" 2024-04-16 OpenJDK Runtime Environment (build 17.0.11+9-Ubuntu-122.04.1) OpenJDK 64-Bit Server VM (build 17.0.11+9-Ubuntu-122.04.1, mixed mode, sharing)); ignoring
configure: (Your Boot JDK version must be one of: 20 21)
checking for javac... /usr/bin/javac
checking for java... /usr/bin/java
configure: Found potential Boot JDK using java(c) in PATH
configure: Potential Boot JDK found at /usr/lib/jvm/java-17-openjdk-amd64 is incorrect JDK version (openjdk version "17.0.11" 2024-04-16 OpenJDK Runtime Environment (build 17.0.11+9-Ubuntu-122.04.1) OpenJDK 64-Bit Server VM (build 17.0.11+9-Ubuntu-122.04.1, mixed mode, sharing)); ignoring
configure: (Your Boot JDK version must be one of: 20 21)
configure: Found potential Boot JDK using well-known locations (in /usr/lib/jvm/openjdk-17)
configure: Potential Boot JDK found at /usr/lib/jvm/openjdk-17 did not contain bin/java; ignoring
configure: Found potential Boot JDK using well-known locations (in /usr/lib/jvm/java-21-openjdk-amd64)
checking for Boot JDK... /usr/lib/jvm/java-21-openjdk-amd64
checking Boot JDK version... openjdk version "21.0.3" 2024-04-16 OpenJDK Runtime Environment (build 21.0.3+9-Ubuntu-1ubuntu122.04.1) OpenJDK 64-Bit Server VM (build 21.0.3+9-Ubuntu-1ubuntu122.04.1, mixed mode, sharing)
checking for java [Boot JDK]... $BOOT_JDK/bin/java
checking for javac [Boot JDK]... $BOOT_JDK/bin/javac
checking for javadoc [Boot JDK]... $BOOT_JDK/bin/javadoc
checking for jar [Boot JDK]... $BOOT_JDK/bin/jar
checking if Boot JDK jar supports --date=TIMESTAMP... true
checking if Boot JDK is 32 or 64 bits... 64
checking for local Boot JDK Class Data Sharing (CDS)... yes, created
checking for Build JDK... yes, will build it for the host platform
...
```
so if there is no need to concern about disk space, install both package would be okay.